Assets/_Game/Scripts/Inventory/InventoryObject.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using UnityEngine;
+using UnityEditor;
+
+[CreateAssetMenu(fileName = "New Inventory", menuName = "DLS/Inventory")]
+public class InventoryObject : ScriptableObject, ISerializationCallbackReceiver
+{
+    public string savePath = "PlayerInventory.save";
+    public List<InventorySlot> container = new List<InventorySlot>();
+    public int currentSize = 0;

```

Comment:
Why is currentSize not just looking at the container.Length?

Reply:
good point I probably should have just remove that variable and use the container.